code/obj/machinery/computer/shuttle.dm

```diff
@@ -314,28 +314,28 @@ var/bombini_saved
 	icon = 'icons/obj/decoration.dmi'
 	icon_state = "syndiepc4"
 
-/obj/machinery/computer/shuttle/emag_act(var/mob/user, var/obj/item/card/emag/E)
-	if(emergency_shuttle.location != SHUTTLE_LOC_STATION)
+/obj/machinery/computer/shuttle/attackby(var/obj/item/W, var/mob/user)
+	if(status & (BROKEN|NOPOWER))
 		return
-	for (var/datum/flock/flock in flocks)
-		if (flock.relay_in_progress)
-			boutput(user, SPAN_ALERT("[src] emits a pained burst of static, but nothing happens!"))
+
+	if (istype(W, /obj/item/disk/data/floppy/read_only/authentication))
```

Comment:
Make this a guard (ie invert the condition and make it return instead of putting the entire proc in an `if` block) and have it call parent, otherwise attacking the shuttle computer normally won't work.

Reply:
I don't quite follow your intent here - can you please say more about the changes you want to see and how'd they would affect the console?

Reply:
Sure, so currently you have the entire proc wrapped in this if statement to check if the item is an auth disk, and if it isn't then nothing happens. So if I hit the shuttle computer with a screwdriver or a fire extinguisher or something it won't defer to the parent `/obj/machinery/computer` behaviour, it'll just do nothing.

An easy way to fix this while also making the code cleaner would be to use [guard pattern](https://en.wikipedia.org/wiki/Guard_(computer_science)) and write something like
```
if (!istype(W, /obj/item/disk/data/floppy/read_only/authentication) || (status & (BROKEN|NOPOWER)))
	return ..()
```
and then leave the rest of the proc to continue below the "guard", having verified that there is power and the item is an auth disk.

Reply:
okay, I'll fix up the entire proc to use this pattern.